app/services/flow/flows/provider_start.rb

```diff
@@ -124,11 +124,18 @@ class ProviderStart < FlowSteps
         link_application_make_links: {
           path: ->(application) { urls.providers_legal_aid_application_link_application_make_link_path(application) },
           forward: lambda do |application|
-            # TODO: This will change when ap-4826 is complete
-            application.proceedings.any? ? :has_other_proceedings : :proceedings_types
+            if application.lead_linked_application.persisted?
+              :link_application_find_link_applications
+            else
+              application.proceedings.any? ? :has_other_proceedings : :proceedings_types
+            end
           end,
           check_answers: :check_provider_answers,
         },
+        link_application_find_link_applications: {
+          path: ->(application) { urls.providers_legal_aid_application_link_application_find_link_application_path(application) },
+          forward: ->(application) { application.proceedings.any? ? :has_other_proceedings : :proceedings_types },
+        },
```

Comment:
Flow refactor: are we refactoring now or  towards the end link case feature?

Reply:
I took the executive decision that we would not include the refactoring on this ticket so as to unblock the downstream work... we can come back and update it later with the original steps